File: scripts/scripts/testing/automations/batch/portugal.R
df <- suppressMessages(readxl::read_excel("input/DATA PORTUGAL.xlsx", sheet = "CUMULATIVE"))
setDT(df)
df <- df[, 1:4]

setnames(df, c("month", "day", "PCR", "Antigen"))

df[, `Cumulative total` := PCR + Antigen]
df[, c("PCR", "Antigen") := NULL]

start_date <- dmy(sprintf("%s %s 2020", df$day[1], df$month[1]))
seq_date <- seq.Date(from = start_date, by = "1 day", length.out = nrow(df))
df$Date <- seq_date

df[, c("day", "month") := NULL]

df <- df[!is.na(`Cumulative total`)]

df[, Country := "Portugal"]
df[, Units := "tests performed"]
df[, `Source URL` := "https://covid19.min-saude.pt/ponto-de-situacao-atual-em-portugal/"]
df[, `Source label` := "Ministry of Health"]
df[, Notes := NA_character_]
df[, `Testing type` := "includes non-PCR"]

fwrite(df, "automated_sheets/Portugal.csv")

File: scripts/scripts/testing/automations/batch/portugal.py
import json
import requests
import pandas as pd


def main():

    url = "https://services5.arcgis.com/eoFbezv6KiXqcnKq/arcgis/rest/services/Covid19_Amostras/FeatureServer/0/query?f=json&where=1%3D1&returnGeometry=false&spatialRel=esriSpatialRelIntersects&outFields=*&orderByFields=Data_do_Relatorio%20asc&resultOffset=0&resultRecordCount=32000&resultType=standard&cacheHint=true"

    data = json.loads(requests.get(url).content)
    data = [elem["attributes"] for elem in data["features"]]

    df = pd.DataFrame.from_records(data)
    df = df[["Total_Amostras__Ac", "Data_do_Relatorio"]].rename(columns={
        "Data_do_Relatorio": "Date", "Total_Amostras__Ac": "Cumulative total"
    })

    df["Date"] = pd.to_datetime(df["Date"], unit="ms")

    df["Country"] = "Portugal"
    df["Units"] = "tests performed"
    df["Source URL"] = "https://covid19.min-saude.pt/ponto-de-situacao-atual-em-portugal/"
    df["Source label"] = "Ministry of Health"
    df["Notes"] = pd.NA
    df["Testing type"] = "includes non-PCR"

    df.to_csv("automated_sheets/Portugal.csv", index=False)


if __name__ == '__main__':
    main()
